From 6c0c0144d891f7685910eedf337623e14004a808 Mon Sep 17 00:00:00 2001
From: Auto Configured <auto.configured>
Date: Wed, 22 Jun 2016 09:29:30 +0200
Subject: [PATCH 1/2] Kconfig: Added CDC option to the DRM kernel config menu

Signed-off-by: Auto Configured <auto.configured>
---
 drivers/gpu/drm/Kconfig | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/Kconfig b/drivers/gpu/drm/Kconfig
index 47f2ce8..a2e3b6c 100644
--- a/drivers/gpu/drm/Kconfig
+++ b/drivers/gpu/drm/Kconfig
@@ -81,6 +81,16 @@ source "drivers/gpu/drm/i2c/Kconfig"
 
 source "drivers/gpu/drm/bridge/Kconfig"
 
+config DRM_TES_CDC
+	bool "TES CDC FB support"
+	depends on DRM
+	select DRM_KMS_CMA_HELPER
+	select DRM_KMS_HELPER
+	select CONFIG_DRM_KMS_HELPER
+	select VIDEOMODE_HELPERS
+	help
+	  Choose this option if you have a CDC display controller.
+
 config DRM_TDFX
 	tristate "3dfx Banshee/Voodoo3+"
 	depends on DRM && PCI
-- 
1.9.1

