From 53ab9fe77548ad40a1f938ca5521e3ac1b4b50ee Mon Sep 17 00:00:00 2001
From: Christian Thaler <christian.thaler@tes-dst.com>
Date: Mon, 4 Jul 2016 13:56:32 +0200
Subject: [PATCH] Added config for fbdev multibuffering

---
 drivers/gpu/drm/Kconfig | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/gpu/drm/Kconfig b/drivers/gpu/drm/Kconfig
index a2e3b6c..563f016 100644
--- a/drivers/gpu/drm/Kconfig
+++ b/drivers/gpu/drm/Kconfig
@@ -77,6 +77,15 @@ config DRM_KMS_CMA_HELPER
 	help
 	  Choose this if you need the KMS CMA helper functions
 
+config DRM_CMA_FBDEV_BUFFER_NUM
+       int "Cma Fbdev Buffer Number"
+       depends on DRM_KMS_CMA_HELPER
+       default 1
+       help
+         Defines the buffer number of cma fbdev.  Default is one buffer.
+         For double buffer please set to 2 and 3 for triple buffer.
+
+
 source "drivers/gpu/drm/i2c/Kconfig"
 
 source "drivers/gpu/drm/bridge/Kconfig"
-- 
1.9.1

